### PR TITLE
Use implicit field mask paths in IS

### PR DESCRIPTION
**Pull request by htdvisser**
**Summary:**

Since #69 (specifically the decisions in https://github.com/TheThingsNetwork/lorawan-stack/issues/69#issuecomment-465184422 and related meeting) we now allow empty field masks. This PR makes sure that the Identity Server handles such requests accordingly.

**Changes:**

- Field Mask paths on IS requests are "cleaned" according to a set of possible/allowed fields, as well as fields to add (default fields on Get-like requests) or remove (ignored fields on Set-like requests)
- CLI is updated to no longer implicitly select the "name" field.

**Notes for Reviewers:**

@rvolosatovs if you think it could be useful in other places (end device registries), we can move the `cleanFieldMaskPaths` utility to `ttnpb`.

**Release Notes: _(optional)_**

The Identity Server now treats empty field masks as if ids and created/updated times were requested.

### CODE REVIEW

**Comment by htdvisser** [comment]
Setting prio/high as this is required for `3.0.0`

File: pkg/identityserver/application_registry.go [hunk 1]
```diff
@@ -83,6 +83,7 @@ func (is *IdentityServer) getApplication(ctx context.Context, req *ttnpb.GetAppl
 	if err = is.RequireAuthenticated(ctx); err != nil {
 		return nil, err
 	}
+	req.FieldMask.Paths = cleanFieldMaskPaths(ttnpb.ApplicationFieldPathsNested, req.FieldMask.Paths, getPaths, nil)
```
Comment (rvolosatovs) [inline pkg/identityserver/application_registry.go]: Why is this necessary?
Why not just set this to `getPaths` if `len(req.FieldMask.Paths) == 0`?
Basically the same question for all other invocations - I really don't get what's going on here.

File: pkg/identityserver/application_registry.go [hunk 2]
```diff
@@ -186,8 +188,14 @@ func (is *IdentityServer) updateApplication(ctx context.Context, req *ttnpb.Upda
 	if err = rights.RequireApplication(ctx, req.ApplicationIdentifiers, ttnpb.RIGHT_APPLICATION_SETTINGS_BASIC); err != nil {
 		return nil, err
 	}
-	if err := validateContactInfo(req.Application.ContactInfo); err != nil {
-		return nil, err
+	req.FieldMask.Paths = cleanFieldMaskPaths(ttnpb.ApplicationFieldPathsNested, req.FieldMask.Paths, nil, getPaths)
+	if len(req.FieldMask.Paths) == 0 {
+		req.FieldMask.Paths = updatePaths
+	}
+	if ttnpb.HasAnyField(req.FieldMask.Paths, "contact_info") {
```
[the remark this answers is not in the document]
Comment (htdvisser) [reply pkg/identityserver/application_registry.go]: 1. `contact_info` is a slice. Field mask paths don't recurse into slices.
2. EndDevice doesn't have a `contact_info` field. For applications and other entities, the allowed paths come from `ttnpb.ApplicationFieldPathsNested` etc.

File: pkg/identityserver/application_registry.go (hunk 1 above)
Comment (htdvisser) [reply pkg/identityserver/application_registry.go]: `getPaths` contains the fields that are implicitly part of the field mask, so they are always added to the field mask in the request, if it doesn't already contain them.

Comment (rvolosatovs) [reply pkg/identityserver/application_registry.go]: Yeah, but there's no harm in adding a field twice - you can just append the fields - we do the same in JS/NS and AS AFAIK.

Comment (rvolosatovs) [reply pkg/identityserver/application_registry.go]: In order to construct the slice of required fields, `Set` registry endpoint in NS constructs a `gets` and `sets`: https://github.com/TheThingsNetwork/lorawan-stack/blob/59154c3b694652b4c72447ae4f74c606003dac80/pkg/networkserver/grpc_deviceregistry.go#L49-L57

Comment (htdvisser) [reply pkg/identityserver/application_registry.go]: This is not the NS/JS. In the IS we prefer to have paths in the field mask only once.

Comment (rvolosatovs) [reply pkg/identityserver/application_registry.go]: But then why not `ttnpb.EnsureFieldpaths(paths []string, ensure ...string) []string`?
You pass the existing fieldmask and the paths you want to add if missing.
You don't need 4 arguments for that.

Comment (htdvisser) [reply pkg/identityserver/application_registry.go]: There is no `ttnpb.EnsureFieldPaths`, I want to combine this with the allowed paths, I want to be able to specify paths that need to be deleted if present, and finally, I want to do it in a single pass instead of combining `EnsureFieldPaths`, `EnsureNotFieldPaths`, `IntersectFieldPaths`, `UniqueFieldPaths`.

Comment (rvolosatovs) [reply pkg/identityserver/application_registry.go]: 1. We have API layer middleware, which should validate that fields that should not be present are not present.
2. Invalid fieldpaths, which still could get through should be validated explicitly and if present - throw an error.

RPC implementation simply should not delete anything passed to it. Cleaning and validation belongs to the API layer middleware. RPC should simply upsert the required fields.
